src/condor_schedd.V6/qmgr_job_updater.cpp

```diff
@@ -80,161 +71,126 @@ QmgrJobUpdater::~QmgrJobUpdater()
 		daemonCore->Cancel_Timer( q_update_tid );
 		q_update_tid = -1;
 	}
-	if( common_job_queue_attrs ) { delete common_job_queue_attrs; }
-	if( hold_job_queue_attrs ) { delete hold_job_queue_attrs; }
-	if( evict_job_queue_attrs ) { delete evict_job_queue_attrs; }
-	if( remove_job_queue_attrs ) { delete remove_job_queue_attrs; }
-	if( requeue_job_queue_attrs ) { delete requeue_job_queue_attrs; }
-	if( terminate_job_queue_attrs ) { delete terminate_job_queue_attrs; }
-	if( checkpoint_job_queue_attrs ) { delete checkpoint_job_queue_attrs; }
-	if( x509_job_queue_attrs ) { delete x509_job_queue_attrs; }
-	delete m_pull_attrs;
 }
 
 
 void
 QmgrJobUpdater::initJobQueueAttrLists( )
 {
-	if( hold_job_queue_attrs ) { delete hold_job_queue_attrs; }
-	if( evict_job_queue_attrs ) { delete evict_job_queue_attrs; }
-	if( requeue_job_queue_attrs ) { delete requeue_job_queue_attrs; }
-	if( remove_job_queue_attrs ) { delete remove_job_queue_attrs; }
-	if( terminate_job_queue_attrs ) { delete terminate_job_queue_attrs; }
-	if( common_job_queue_attrs ) { delete common_job_queue_attrs; }
-	if( checkpoint_job_queue_attrs ) { delete checkpoint_job_queue_attrs; }
-	if( x509_job_queue_attrs ) { delete x509_job_queue_attrs; }
-	delete m_pull_attrs;
-
-	common_job_queue_attrs = new StringList();
```

Comment:
Can we just assign all these in one go, instead of calling "insert" a milion times, e.g. something like

```
common_job_queue_attrs = {
   ATTR_ONE_FISH,
   ATTR_TWO_FISH,
   ATTR_RED_FISH,
   ATTR_BLUE_FISH
};
```

Reply:
I believe making them const would require putting the arrays of values in the initializer list of the constructor. I don't like making the initializer list that long. But I like getting rid of the million inserts.